### PR TITLE
Feature/iphone x support

### PR DESCRIPTION
**Pull request by RaimundasSakalauskas**
One more huge PR. This is low priority compared to the other 2. This replaces dated dependencies and fixes some app bugs.

### CODE REVIEW

File: Photon-Tinker/DeviceInspectorDataViewController.swift [hunk 1]
```diff
@@ -56,7 +55,7 @@ class DeviceInspectorDataViewController: DeviceInspectorChildViewController, UIT
     
     
     override func viewWillDisappear(_ animated: Bool) {
-        IQKeyboardManager.shared().shouldHidePreviousNext = false
+        IQKeyboardManager.shared().previousNextDisplayMode = .default
```
Comment (idokleinman) [inline Photon-Tinker/DeviceInspectorDataViewController.swift:58]: Think we should continue using IQKeyboardManager? thought you didn't like it

File: Photon-Tinker/SPKTinkerViewController.m [hunk 2]
```diff
@@ -597,9 +595,9 @@ - (void)pinCallHome:(PinView *)pinView
             [pinView endUpdating];
             
             NSString* errorStr = [NSString stringWithFormat:@"Error communicating with device (%@)",errorMessage];
-            [[Mixpanel sharedInstance] track:@"Tinker: error" properties:@{@"type":@"communicate with device"}];
+            [[SEGAnalytics sharedAnalytics] track:@"Tinker: error" properties:@{@"type":@"communicate with device"}];
```
Comment (idokleinman) [inline Photon-Tinker/SPKTinkerViewController.m:598]: Did we check that analytics events are flowing correctly now?

File: Photon-Tinker/ScanBarcodeViewController.m [hunk 3]
```diff
@@ -100,15 +104,32 @@ - (void)viewDidLoad
...
+    [self.view bringSubviewToFront:_cancelButton];
 
     
 }
 
 
+- (void)viewWillLayoutSubviews {
+    [super viewWillLayoutSubviews];
+
+    if (@available(iOS 11.0, *)) {
```
Comment (idokleinman) [inline Photon-Tinker/ScanBarcodeViewController.m:119]: Will it look ok on <iOS 11 devices?

File: Photon-Tinker/DeviceInspectorDataViewController.swift (hunk 1 above)
Comment (RaimundasSakalauskas) [reply Photon-Tinker/DeviceInspectorDataViewController.swift:58]: I didn't like the way keyboard behaved. We thought the reason for that was IQKeyboardManager, it turns out it was just a problem with configuration and some usage limitations which I believe I was able to overcome.

File: Photon-Tinker/ScanBarcodeViewController.m (hunk 3 above)
Comment (RaimundasSakalauskas) [reply Photon-Tinker/ScanBarcodeViewController.m:119]: Yes, this only modifies layout for devices having safeArea (which is only available in ios 11). Original layout code is still being run for older iOS versions.

File: Photon-Tinker/SPKTinkerViewController.m (hunk 2 above)
Comment (RaimundasSakalauskas) [reply Photon-Tinker/SPKTinkerViewController.m:598]: Checking with jberi.

Comment (RaimundasSakalauskas) [reply Photon-Tinker/SPKTinkerViewController.m:598]: Seems to be working in Segment, need you to check in Firebase.

**Comment by RaimundasSakalauskas** [comment]
Can this be merged or do you @idokleinman want to have one more look at it? I have finished everything I was willing to do within this PR.

**Pull request merged**